src/ClientImpl.cpp

```diff
@@ -644,6 +785,44 @@ class SyncCallback : public ProcedureCallback {
     InvocationResponse *m_responseOut;
 };
 
+void ClientImpl::purgeExpiredRequests() {
+    struct timeval now;
```

Comment:
I think this is thread safe as the purging will happen in the same thread context that pushes queue and deques callbacks.